packages/x-date-pickers-pro/src/DateRangeCalendar/useDragRange.ts

```diff
@@ -95,87 +100,138 @@ const useDragRangeEvents = <TDate>({
       'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';
   }, []);
 
+  const isElementDraggable = (day: TDate | null): day is TDate => {
+    if (day == null) {
+      return false;
+    }
+
+    const shouldInitDragging = !disableDragEditing && !!dateRange[0] && !!dateRange[1];
+    const isSelectedStartDate = isStartOfRange(utils, day, dateRange);
+    const isSelectedEndDate = isEndOfRange(utils, day, dateRange);
+
+    return shouldInitDragging && (isSelectedStartDate || isSelectedEndDate);
+  };
+
   const handleDragStart = useEventCallback((event: React.DragEvent<HTMLButtonElement>) => {
-    event.stopPropagation();
     const newDate = resolveDateFromTarget(event.target, utils);
-    if (newDate) {
-      if (emptyDragImgRef.current) {
-        event.dataTransfer.setDragImage(emptyDragImgRef.current, 0, 0);
-      }
-      setRangeDragDay(newDate);
-      event.dataTransfer.effectAllowed = 'move';
-      setIsDragging(true);
-      const buttonDataset = (event.target as HTMLButtonElement).dataset;
-      if (buttonDataset.timestamp) {
-        event.dataTransfer.setData('draggingDate', buttonDataset.timestamp);
-      }
-      if (buttonDataset.position) {
-        onDatePositionChange(buttonDataset.position as DateRangePosition);
-      }
+    if (!isElementDraggable(newDate)) {
+      return;
+    }
+
+    event.stopPropagation();
+    if (emptyDragImgRef.current) {
+      event.dataTransfer.setDragImage(emptyDragImgRef.current, 0, 0);
+    }
+    setRangeDragDay(newDate);
+    event.dataTransfer.effectAllowed = 'move';
+    setIsDragging(true);
+    const buttonDataset = (event.target as HTMLButtonElement).dataset;
+    if (buttonDataset.timestamp) {
+      event.dataTransfer.setData('draggingDate', buttonDataset.timestamp);
+    }
+    if (buttonDataset.position) {
+      onDatePositionChange(buttonDataset.position as DateRangePosition);
     }
   });
+
   const handleTouchStart = useEventCallback((event: React.TouchEvent<HTMLButtonElement>) => {
     const target = resolveElementFromTouch(event);
     if (!target) {
       return;
     }
+
     const newDate = resolveDateFromTarget(target, utils);
-    if (newDate) {
-      setRangeDragDay(newDate);
-      setIsDragging(true);
-      const button = event.target as HTMLButtonElement;
-      const buttonDataset = button.dataset;
-      if (buttonDataset.position) {
-        onDatePositionChange(buttonDataset.position as DateRangePosition);
-      }
+    if (!isElementDraggable(newDate)) {
+      return;
+    }
+
+    setRangeDragDay(newDate);
+    setIsDragging(true);
+    const button = event.target as HTMLButtonElement;
+    const buttonDataset = button.dataset;
+    if (buttonDataset.position) {
+      onDatePositionChange(buttonDataset.position as DateRangePosition);
     }
   });
+
   const handleDragEnter = useEventCallback((event: React.DragEvent<HTMLButtonElement>) => {
+    if (!isDragging) {
```

Comment:
I have specifically avoided using `isDragging` in these functions to avoid having the need to re-create them whenever that flag changes. 🤔 

But if you think it's the only way to achieve performance gains we need—it probably doesn't matter that much as compared to the reduction in computations. 🙈 👍

Reply:
I think that this question is [answered here](https://github.com/mui/mui-x/pull/7066#discussion_r1039251650) 👌